oneclick.py

```diff
@@ -856,6 +867,9 @@ def update_metadata(self, catalog_item, identifier=None, metadata_replacement_po
             identifier, metadata, metadata_replacement_policy=metadata_replacement_policy
         )
 
+        if not isinstance(result, CoverageFailure):
+            self.handle_success(identifier)
+
```

Comment:
I would expect `handle_success` to be called by the code that calls `process_item`, like `BibliographicCoverageProvider.process_batch`. I get why you're doing this -- you don't call `process_batch` because OneClick data isn't batched. You get the whole thing at once so you might as well process it all at once, in `populate_all_catalog`.

But in `populate_all_catalog` you already have an `if not isinstance(result, CoverageFailure):`. How about calling `handle_success` there? Then you could get rid of this, and if you ever do call `process_batch` then `handle_success` won't be called twice.